packages/@aws-cdk/aws-apigateway/lib/method.ts

```diff
@@ -360,7 +360,12 @@ export class Method extends Resource {
     }
 
     if (options.requestValidatorOptions) {
-      const validator = (this.api as RestApi).addRequestValidator('validator', options.requestValidatorOptions);
+      let id = `${this.resource.path}${this.httpMethod}Validator`;
+      // For backward compatibility
+      if (options.requestValidatorOptions.requestValidatorName) {
+        id = 'validator';
+      }
+      const validator = (this.api as RestApi).addRequestValidator(id, options.requestValidatorOptions);
```

Comment:
We should only call `addRequestValidator()` inside the `if (options.requestValidatorOptions)` block because outside of that block, the caller has specified an actual request validator. If they've specified one, we don't want to create a new one; instead, we just want to return the `requestValidatorId`.